components/sidebar/sidebar_service.cc

```diff
@@ -533,25 +549,13 @@ void SidebarService::LoadSidebarItems() {
   }
 
   // Add the items the user has never seen (or never persisted).
-  // Get the initial order of items so that we can attempt to
-  // insert at the intended order.
+  // Insert just behind the last built-in item.
   for (const auto& item : default_items_to_add) {
-    const auto* default_item_iter = base::ranges::find(
-        SidebarService::kDefaultBuiltInItemTypes, item.built_in_item_type);
-    auto default_index = default_item_iter -
-                         std::begin(SidebarService::kDefaultBuiltInItemTypes);
-    // Add at the default index for the first time. For users which haven't
-    // changed any order, or removed items, this will be at the intentional
-    // index. For users who have re-ordered, this will be different but still
-    // acceptable. It will be a minority of cases where it gets inserted in to
-    // the middle of custom items, but that will still work.
-    auto index = std::min(static_cast<int>(default_index),
-                          static_cast<int>(items_.size()));
+    const size_t pos_to_insert = GetLastBuiltInItemIndex(items());
```

Comment:
Yeah I'm afraid I don't think this is what we want to do in this function as this would mean any new item we add would always be at the end for existing users. And we want to insert new items at a predictable index, especially for the majority users who haven't customized sidebar items.

Reply:
removed that